databaseAnnotator/src/main/java/org/cbioportal/database/annotator/AnnotateRecordsProcessor.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.cbioportal.database.annotator;
+
+import java.sql.*;
+import com.querydsl.sql.SQLQueryFactory;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.log4j.Logger;
+import org.cbioportal.models.*;
+import org.cbioportal.annotator.Annotator;
+import org.cbioportal.database.annotator.model.*;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.client.*;
+
+/**
+ *
+ * @author heinsz
+ */
+public class AnnotateRecordsProcessor implements ItemProcessor<MutationEvent, MutationEvent>{
+    @Value("#{jobParameters[isoform]}")
+    private String isoform;
+
+    @Autowired
+    Annotator annotator;
+
+    @Autowired
+    SQLQueryFactory databaseAnnotatorQueryFactory;
+
+    public final static int NA_INT = -1;
+
+    private final Logger log = Logger.getLogger(AnnotateRecordsProcessor.class);
+
+    @Override
+    public MutationEvent process(MutationEvent i) throws Exception {
+        // first make a MutationRecord that we can send to Genome Nexus to annotate
+        Map<String, String> mafLine = new HashMap<>();
+        String chromosome;
+        if (i.getCHR().equals("23")) {
+                chromosome = "X";
+        }
+        else if (i.getCHR().equals("24")) {
+                chromosome = "Y";
+        }
+        else {
+                chromosome = i.getCHR();
+        }
+        mafLine.put("Chromosome", chromosome);
+        mafLine.put("Start_Position", String.valueOf(i.getSTART_POSITION()));
+        mafLine.put("End_Position", String.valueOf(i.getEND_POSITION()));
+        mafLine.put("Reference_Allele", i.getREFERENCE_ALLELE());
+        mafLine.put("Tumor_Seq_Allele2", i.getTUMOR_SEQ_ALLELE());
+        MutationRecord record = annotator.createRecord(mafLine);
+        AnnotatedRecord annotatedRecord;
+        try {
+                annotatedRecord = annotator.annotateRecord(record, true, isoform, true);
+        }
+        catch (HttpServerErrorException | ResourceAccessException e) {
+           log.error("Failed to annotate record - errors accessing Genome Nexus");
+           return i;
+        }
+
+        // If the mutation is a 5' or 3' flank mutation, or any other UTR, it shouldn't be in the database as these are supposed to be filtered.. Remove it.
+        if (annotatedRecord.getVariant_Classification().equals("5'Flank") || 
```

Comment:
we only want to ignore 5'Flank mutations if the gene is *_not_* TERT 

ref: https://github.com/cBioPortal/cbioportal/blob/c628bfc06e0f7446b4c2a51892f7acc8b6087325/core/src/main/java/org/mskcc/cbio/portal/scripts/MutationFilter.java#L142